### PR TITLE
Collision d2

### PR DESCRIPTION
**Pull request by brb1031**
Makes all collision detection depend on distance (rather than d^2 or
d^3), to agree with viewport’s description text.  Resolves issue #27

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,32 +1,33 @@
-# Windows image file caches
```
Comment (mcw0933) [inline .gitignore]: This diff is potentially the result of line-ending settings.  I don't see any actual content change here.

There are a couple ways to correct this.  One is we could all configure git locally to use some particular line ending style for this repo.

The other is we could add a .gitattributes file that defines what type of line endings each file type in the repo should have.  That's the route I ended up taking with another side project where we have Mac and Windows devs.

Docs: https://help.github.com/articles/dealing-with-line-endings/#platform-all

**Comment by mcw0933** [comment]
This all seems legit.  Unfortunately since it's stacked on top of the previous changes it's hard to see what's new here.  Looks like most of this is not-yet-merged, dependency code from the previous PR.

@brb1031 - Is there a branch other than master that we could compare this to, to see just the diff for this feature?

**Comment by brb1031** [comment]
Hm.  I deliberately branched from master so that there would only be a few one-line changes.  But this contains all kinds of stuff from omit-negligible, which shouldn't have been in my local files at all....

There must be some super-basic confusion I'm having with checking out branches...

**Comment by brb1031** [comment]
I'm going to kill this one, and branch again.

**Pull request closed without merging**